### PR TITLE
fix: New Booker: Success Redirect for Recurring as well non recurring bookings

### LINKED ISSUE #9587: [CAL-1975] new booker: redirect not working
Redirect after booking seems to have broken.   Try booking on <https://cal.com/bmotters/system-review>.    This is set up to redirect to <https://dev.riverscape.info/booking/calcom/success2.html>.    Was previously working.

[Slack Message](https://calendso.slack.com/archives/C01NXDSRY9W/p1686948884344819?thread_ts=1686948884.344819&cid=C01NXDSRY9W)

<sub>[CAL-1975](https://linear.app/calcom/issue/CAL-1975/new-booker-redirect-not-working)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?
Fixes #9587

New Booker had missing success redirect URL logic.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Tested following
- [x] Recurring Event Booking 
	- [x] with Success Redirect URL set to https://google.com
	- [x] With no success redirect URL
- [x] Non Recurring Event Booking 
	- [x] with Success Redirect URL set to https://google.com
	- [x] With no success redirect URL

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx [hunk 1]
```diff
@@ -194,17 +172,20 @@ export const BookEventForm = ({ onCancel }: BookEventFormProps) => {
...
-          uid,
-          email: bookingForm.getValues("responses.email"),
-          formerTime: rescheduleBooking?.startTime
-            ? dayjs(rescheduleBooking?.startTime).toISOString()
-            : undefined,
-          slug: `${eventSlug}`,
-          isRecurring: false,
-        })
-      );
+      const query = {
```
Comment (hariombalhara) [inline packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx:175]: Logic borrowed from https://github.com/calcom/cal.com/blob/77e9c4a8c6a02aef45a409441dda2d37de06095e/apps/web/components/booking/pages/BookingPage.tsx#L281

**Review by hariombalhara — COMMENTED** [review]
Self reviewed

**Review by PeerRich — APPROVED** [review]

**Comment by ericrommel** [comment]
There is a bug with the number of events when it is equal to or less than 5. Issue created: https://github.com/calcom/cal.com/issues/9604

**Review by ericrommel — DISMISSED** [review]
https://github.com/calcom/cal.com/issues/9604

**Comment by hariombalhara** [comment]
@ericrommel Thanks for thorough testing 🙏 
It is an unrelated issue that I am sure is already on production. So, it shouldn't block this PR. 

For this high priority fix, can you confirm success redirect URL is working.

**Comment by PeerRich** [comment]
image
getting this on vercel preview somehow

**Comment by PeerRich** [comment]
tested. works!

**Pull request merged**